csrc/multidevice/communication.cpp

```diff
@@ -210,6 +210,53 @@ std::string Communication::toInlineString(int indent_size) const {
   return toString(indent_size);
 }
 
+std::ostream& operator<<(std::ostream& os, const P2PCommunicationType& type) {
+  switch (type) {
+    case P2PCommunicationType::send:
+      os << "send";
+      break;
+    case P2PCommunicationType::recv:
+      os << "recv";
+      break;
+    default:
+      NVF_THROW("unrecognized P2PCommunicationType: ", type);
+  }
+  return os;
+}
+
+P2PCommunication::P2PCommunication(
+    IrBuilderPasskey passkey,
+    P2PCommunicationType type,
+    TensorView* buffer,
+    Val* peer,
+    Val* tag)
```

Comment:
What's tag for?

Reply:
tag is to differentiate between multiple sends and recv between the same peers. So that if one rank makes two sends to another, the send can be identified by tag-matching.

However, NCCL doesn't support tags. So I would be ok to remove it for now, but still, we might use it in the future.

Reply:
Thanks! I recommend we remove it for simplicity.

Reply:
ok, done